server/src/test/java/org/opensearch/search/aggregations/bucket/terms/TermsAggregatorTests.java

```diff
@@ -307,38 +332,63 @@ public void testSimple() throws Exception {
         }
     }
 
+    /**
+     * This test case utilizes the default implementation of GlobalOrdinalsStringTermsAggregator.
+     * In this case, the segment terms will not get initialized and will run without LeafBucketCollector#termDocFreqCollector optimization
+     */
     public void testStringIncludeExclude() throws Exception {
+        testStringIncludeExclude(
+            (document, field, value) -> document.add(new SortedSetDocValuesField(field, new BytesRef(value))),
+            DEFAULT_POST_COLLECTION
+        );
+    }
+
+    /**
+     * This test case utilizes the default implementation of GlobalOrdinalsStringTermsAggregator.
+     * In this case, the segment terms will get initialized and will use LeafBucketCollector#termDocFreqCollector optimization
+     */
+    public void testStringIncludeExcludeWithStoredValues() throws Exception {
+        // aggregator.postCollection() is not required when LeafBucketCollector#termDocFreqCollector optimization is used
+        // This also verifies that the bucket count is completed without running postCollection()
+        testStringIncludeExclude((document, field, value) -> {
+            document.add(new SortedSetDocValuesField(field, new BytesRef(value)));
+            document.add(new StringField(field, value, Field.Store.NO));
```

Comment:
It's a little confusing here. `WithStoredValues` -> `new StringField(field, value, Field.Store.NO)`, is this just to index the term without storing the value? so `WithIndex` is probably better. Same for other places

Reply:
@bowenlan-amzn I think your doubt is regarding the usage of `Field.Store.NO` which gives the impression of whether the value is stored or not.

Here is the snippet from Lucene documentation of [StringField](https://lucene.apache.org/core/7_3_1/core/org/apache/lucene/document/StringField.html) constructor:

```
public StringField(String name,
                   String value,
                   Field.Store stored)
Creates a new textual StringField, indexing the provided String value as a single token.

Parameters:
name - field name
value - String value
stored - Store.YES if the content should also be stored

Throws:
IllegalArgumentException - if the field name or value is null.
```

Lucene I guess has made it confusing, this could have been named as something as `Content` to make it clear.


However, I think I will take your suggestion and rename `StoredValues` to `IndexedValues` to make it clear. Does it makes it more explanatory?

Reply:
Incidentally, the easiest thing is probably to add a single `KeywordField`, since that's essentially a `StringField` with doc values.

Reply:
Thanks @msfroh for the suggestion. I am making the changes in subsequent commit.